src/io/flutter/run/daemon/DeviceDaemon.java

```diff
@@ -262,6 +263,11 @@ public void onDaemonLogMessage(@NotNull DaemonEvent.LogMessage message) {
       LOG.info("flutter device watcher: " + message.message);
     }
 
+    @Override
+    public void onDaemonShowMessage(@NotNull DaemonEvent.ShowMessage event) {
+      FlutterMessages.showMessage(event.title, event.message);
```

Comment:
The intent of the `showMessage` event is that clients can choose how to render it - most IDE clients would likely translate it into a toast. Perhaps this could be make more clear by docs on the flutter_tools side - to indicate that `showMessage` will have a client effect (I tried to make that pretty clear in the wiki protocol docs: https://github.com/flutter/flutter/wiki/The-flutter-daemon-mode#showmessage)